vipermonkey/core/expressions.py

```diff
@@ -2088,12 +2109,19 @@ def _handle_regex_execute(self, context, tmp_lhs):
             return None
 
         # Find all the regex matches in the string.
-        r = None
         try:
-            r = re.findall(pat, mod_str)
+            matches = re.finditer(pat, mod_str)
+            match_objects = []
+            for match in matches:
+                # create RegexResult so that ViperMonkey can access fields like FirstIndex
+                match_object = RegexResult(str(match), int(match.start()))
+                match_objects.append(match_object)
```

Comment:
This whole loop can be replaced with a list comprehension `return [RegexResult(str(match), int(match.start())) for match in re.finditer(pat, mod_str)]`.